### PR TITLE
Make the testing scripts executable on Python 3.x 

### LINKED ISSUE #7094: `tools/run_tests/run_tests.py` does not work with Python 3
The `run_tests.py` script does not work with python 3 and throws the following error.
```
Password:
  File "run_tests.py", line 203
    print '\nWARNING: binary not found, skipping', binary
                                                ^
SyntaxError: Missing parentheses in call to 'print'
```
Hopefully, we can fix this incompatibility by moving the print statements to `print()` functions. This was one of the breaking changes in Python3. Most likely: Fixing `print` makes it Python3.x compatible. Worst case: There might be still be some incompatibilities but they should be easy enough to fix. 

As there's this whole move towards making grpc Python 3.x compliant, I highly suggest abstaining from writing scripts that only work with Python 2.x and not with Python 3.x. For all practical purposes, that requires using the `print()` function instead of the statements and using new style Classes (i.e. `class Foo(object):` instead of `class Foo:`) in python. I'll create a PR to solve the current issue with running tests with python3 in the meanwhile.

### PR DESCRIPTION
**Pull request by thunderboltsid**
Fixes #7094 
@jtattermusch please review.

### CODE REVIEW

**Comment by jtattermusch** [comment]
In general, I think it looks fine but it is super important to make sure that all the tests are passing (because run_tests.py changes have the potential to break everything).

One of the problems I saw:
```
Traceback (most recent call last):
  File "tools/run_tests/run_tests.py", line 49, in <module>
    from six.moves import urllib
ImportError: No module named six.moves
```

Also, needs resolving.

**Comment by jtattermusch** [comment]
Reassigned to @kpayson64  because I'll be OOO soon.

File: tools/run_tests/port_server.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 """Manage TCP ports for unit tests; started by run_tests.py"""
 
 import argparse
-import BaseHTTPServer
+from six.moves import BaseHTTPServer
```
Comment (jtattermusch) [inline tools/run_tests/port_server.py]: this breaks lot of stuff.

**Comment by soltanmm** [comment]
Note: the test machines (the dockerfiles for the test machines) need to get `six` installed on them.

**Comment by thunderboltsid** [comment]
Okay, since this is flooded by a bunch of changes after generate_projects.sh ran, I'm create another PR updated with the main fork head and will split the commits until at least the code review.

**Pull request closed without merging**